server/odc-service/src/main/java/com/oceanbase/odc/service/flow/listener/ApprovalStatusNotifyListener.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.flow.listener;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.oceanbase.odc.core.flow.util.EmptyExecutionListener;
+import com.oceanbase.odc.metadb.flow.FlowInstanceApprovalViewEntity;
+import com.oceanbase.odc.metadb.flow.FlowInstanceApprovalViewRepository;
+import com.oceanbase.odc.metadb.iam.resourcerole.UserResourceRoleEntity;
+import com.oceanbase.odc.metadb.iam.resourcerole.UserResourceRoleRepository;
+import com.oceanbase.odc.metadb.task.TaskEntity;
+import com.oceanbase.odc.service.flow.FlowInstanceService;
+import com.oceanbase.odc.service.flow.FlowableAdaptor;
+import com.oceanbase.odc.service.flow.instance.FlowApprovalInstance;
+import com.oceanbase.odc.service.notification.Broker;
+import com.oceanbase.odc.service.notification.NotificationProperties;
+import com.oceanbase.odc.service.notification.helper.EventBuilder;
+import com.oceanbase.odc.service.notification.model.Event;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author liuyizhuo.lyz
+ * @date 2024/2/26
+ */
+@Slf4j
+public class ApprovalStatusNotifyListener extends EmptyExecutionListener {
+
+    @Autowired
+    private NotificationProperties notificationProperties;
+    @Autowired
+    private EventBuilder eventBuilder;
+    @Autowired
+    private Broker broker;
+    @Autowired
+    private FlowInstanceService flowInstanceService;
+    @Autowired
+    private FlowableAdaptor flowableAdaptor;
+    @Autowired
+    FlowInstanceApprovalViewRepository flowInstanceApprovalViewRepository;
+    @Autowired
+    UserResourceRoleRepository userResourceRoleRepository;
+
+    @Override
+    protected void onExecutiuonStart(DelegateExecution execution) {
+        if (!notificationProperties.isEnabled()) {
+            return;
+        }
+        try {
+            FlowApprovalInstance target = getTarget(execution);
+            TaskEntity taskEntity = flowInstanceService.getTaskByFlowInstanceId(target.getFlowInstanceId());
```

Comment:
judge if approval inst is auto-approved and then `flowInstanceService.getTaskByFlowInstanceId(target.getFlowInstanceId());`

Reply:
done